### PR TITLE
Feature/provault poc

### PR DESCRIPTION
**Pull request by LaurensKubat**
## 4. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

**Review by LaurensKubat — COMMENTED** [review]
Left initial comments on the layout/architecture of the vault

File: smart-contracts/contracts/pro-vault/src/position_manager/vault_position_manager.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+use crate::error::ContractError;
+use serde::{Serialize, Deserialize};
+use schemars::JsonSchema;
+
+pub const PROVAULT_POSITION: Item<ProVaultPosition> = Item::new("provault_position");
+
+// #[cw_serde]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct ProVaultPosition {
+    pub provault_share_balance: Vec<Coin>,  // Total pro vault share
```
[the remark this answers is not in the document]
Comment (akure) [reply smart-contracts/contracts/pro-vault/src/position_manager/vault_position_manager.rs:12]: Yes. Total amount of minted shares by the pro's vault.  At any point in time, it should be  [ Total minted on deposits - Total burned on withdrawals ]

File: smart-contracts/contracts/pro-vault/src/position_manager/vault_position_manager.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    pub last_updated: u64,
+    pub total_unallocated: Uint128,  // Remaining amount of deposited token
```
[the remark this answers is not in the document]
Comment (akure) [reply smart-contracts/contracts/pro-vault/src/position_manager/vault_position_manager.rs:14]: What will be the use of unallocated deposit tokens when we keep it on the vault contract?  One obvious option is to send that amount for staking.

File: smart-contracts/contracts/pro-vault/src/yield_manager/adaptor_yield.rs
[the remark this answers is not in the document]
Comment (akure) [reply smart-contracts/contracts/pro-vault/src/yield_manager/adaptor_yield.rs:1]: Like any yield we generate from multiple adaptors will be tracked here on demand for queries. So could be use only for organizing the queries. And we don't actually store anything on it. 

Adaptor 1 -> yield 1 
Adaptor 2 -> yield 2 
Adaptor 3 -> yield 3

**Pull request closed without merging**